nailgun/entities.py

```diff
@@ -1327,6 +1328,121 @@ def clone(self, synchronous=True, **kwargs):
         return _handle_response(response, self._server_config, synchronous)
 
 
+class TemplateInput(
+    Entity,
+    EntityCreateMixin,
+    EntityDeleteMixin,
+    EntityReadMixin,
+    EntitySearchMixin,
+    EntityUpdateMixin
+):
+    """A representation of a Template Input entity."""
+    def __init__(self, server_config=None, **kwargs):
+        _check_for_value('template', kwargs)
+        self._fields = {
+            'advanced': entity_fields.BooleanField(),
+            'description': entity_fields.StringField(),
+            'fact_name': entity_fields.StringField(),
+            'input_type': entity_fields.StringField(),
+            'name': entity_fields.StringField(),
+            'options': entity_fields.StringField(),
+            'puppet_class_name': entity_fields.StringField(),
+            'puppet_parameter_name': entity_fields.StringField(),
+            'required': entity_fields.BooleanField(),
+            # There is no Template base class yet
+            'template': entity_fields.OneToOneField(
+                JobTemplate, required=True),
+            'variable_name': entity_fields.StringField(),
+        }
+        super(TemplateInput, self).__init__(server_config, **kwargs)
+        self._meta = {
+            'api_path': '/api/v2/templates/{0}/template_inputs'
+            .format(self.template.id),
+            'server_modes': ('sat')
+        }
+
+    def read(self, entity=None, attrs=None, ignore=None, params=None):
+        """Create a JobTemplate object before calling read()
+        ignore 'advanced'
+        """
+        if entity is None:
+            entity = TemplateInput(template=self.template)
```

Comment:
The test are failing because of this line , the server config is missing:
```python
 entity = TemplateInput(self._server_config, template=self.template)
```
Please correct

Reply:
Please also rebase

Reply:
Thanks, that did the trick!
I just rebased and squashed everything.